client/client.go

```diff
@@ -0,0 +1,1375 @@
+	return request.Client.WriteExecute(request)
+}
+
+func (client *OpenFgaClient) WriteExecute(request SdkClientWriteRequest) (ClientWriteResponse, error) {
+	var maxPerChunk = int32(1) // 1 has to be the default otherwise the chunks will be sent in transactions
+	if request.options != nil && request.options.Transaction != nil {
+		maxPerChunk = request.options.Transaction.MaxPerChunk
+	}
+	var maxParallelReqs = DEFAULT_MAX_METHOD_PARALLEL_REQS
+	if request.options != nil && request.options.Transaction != nil {
+		maxParallelReqs = request.options.Transaction.MaxParallelRequests
+	}
+
+	if request.options == nil || request.options.Transaction == nil || !request.options.Transaction.Disable {
+		writeRequest := openfga.WriteRequest{
+			AuthorizationModelId: client.getAuthorizationModelId(request.options.AuthorizationModelId),
+		}
+		if request.body.Writes != nil && len(*request.body.Writes) > 0 {
+			writes := openfga.TupleKeys{}
+			for index := 0; index < len(*request.body.Writes); index++ {
+				writes.TupleKeys = append(writes.TupleKeys, (*request.body.Writes)[index].ToTupleKey())
+			}
+			writeRequest.Writes = &writes
+		}
+		if request.body.Deletes != nil && len(*request.body.Deletes) > 0 {
+			deletes := openfga.TupleKeys{}
+			for index := 0; index < len(*request.body.Deletes); index++ {
+				deletes.TupleKeys = append(deletes.TupleKeys, (*request.body.Deletes)[index].ToTupleKey())
+			}
+			writeRequest.Deletes = &deletes
+		}
+		_, httpResponse, err := client.OpenFgaApi.Write(request.ctx).Body(writeRequest).Execute()
+
+		response := ClientWriteResponse{Writes: []ClientWriteSingleResponse{{}}, Deletes: []ClientWriteSingleResponse{{}}}
+
+		clientWriteStatus := SUCCESS
+		if err != nil {
+			clientWriteStatus = FAILURE
+		}
+
+		for index := 0; index < len(writeRequest.Writes.GetTupleKeys()); index++ {
+			response.Writes = append(response.Writes, ClientWriteSingleResponse{
+				TupleKey: (*request.body.Writes)[index],
+				Response: httpResponse,
+				Status:   clientWriteStatus,
+				Error:    err,
+			})
+		}
+		for index := 0; index < len(writeRequest.Deletes.GetTupleKeys()); index++ {
+			response.Deletes = append(response.Deletes, ClientWriteSingleResponse{
+				TupleKey: (*request.body.Deletes)[index],
+				Response: httpResponse,
+				Status:   clientWriteStatus,
+				Error:    err,
+			})
+		}
+		return response, err
+	}
+
+	var writeChunkSize = int(maxPerChunk)
+	var writeChunks [][]ClientTupleKey
+	for i := 0; i < len(*request.body.Writes); i += writeChunkSize {
+		end := i + writeChunkSize
+
+		if end > len(*request.body.Writes) {
+			end = len(*request.body.Writes)
+		}
+
+		writeChunks = append(writeChunks, (*request.body.Writes)[i:end])
+	}
+
+	writeGroup, ctx := errgroup.WithContext(request.ctx)
+	writeGroup.SetLimit(int(maxParallelReqs))
+	writeResponses := make([]ClientWriteResponse, len(writeChunks))
+	for index, writeBody := range writeChunks {
+		index, writeBody := index, writeBody
```

Comment:
the naming of these variables make things a bit confusing

Reply:
`writeChunks` is the array that was created from the writes send in the body (e.g. `[[a,b],[c,d],[e]]`)
`writeBody` is the part needed for this specific write (e.g. `[a,b]`)